ecl/hthor/hthor.cpp

```diff
@@ -8446,8 +8496,7 @@ bool CHThorDiskReadBaseActivity::openNext()
             unsigned numCopies = curPart?curPart->numCopies():ldFile->numPartCopies(partNum);
             //MORE: Order of copies should be optimized at this point....
             StringBuffer file, filelist;
-            closepart();
-            if (dfsParts && superfile && curPart)
+            if (dfsParts && superfile && curPart && !subfileLogicalFilenames.empty())
```

Comment:
Has the closepart() moved to the correct place?  Would it also work to call closepart() after the while loop?  Asking because there is another call to closepart() within the loop code, and not sure exactly why the change was needed (to make sure stats were updated?)

Reply:
I realize that it's not actually reading the file within that loop.  I've moved closepart back to the original location.